### PR TITLE
add suggestion to use cache if no depot is detected

### PR DESCRIPTION
**Pull request by IanButterworth**
Closes https://github.com/julia-actions/julia-buildpkg/issues/40

Adds a suggestion to use `julia-actions/cache` if no depot is detected.

Screenshot 2023-11-21 at 2 50 00 PM

Can be disabled with `ignore-no-cache: true`

If this is approved, I'll do the same for `julia-runtest` (whichever comes first in a run will show the note then create the depot, so we won't end up with double notifications)

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -21,6 +21,9 @@ inputs:
     description: 'Determine if Pkg uses the cli git executable (Julia >= 1.7). Might be necessary for more complicated SSH setups.
                   Options: true | false. Default : false'
     default: 'false'
+  ignore-no-cache:
+    description: 'Whether to ignore if there appears to be no depot caching. Silences an action warning.'
+    default: 'no'
```
Comment (IanButterworth) [inline action.yml:26]:
```suggestion
    default: 'false'
```

File: action.yml [hunk 2]
```diff
@@ -29,8 +32,11 @@ runs:
     run: echo "JULIA_PKG_SERVER_REGISTRY_PREFERENCE=${JULIA_PKG_SERVER_REGISTRY_PREFERENCE:-eager}" >> ${GITHUB_ENV}
     shell: bash
   - run: |
+      if "${{ inputs.ignore-no-cache }}" == "no" && !isdir(DEPOT_PATH[1])
```
Comment (IanButterworth) [inline action.yml:35]:
```suggestion
      if "${{ inputs.ignore-no-cache }}" == "false" && !isdir(DEPOT_PATH[1])
```

File: action.yml [hunk 3]
```diff
@@ -29,8 +32,11 @@ runs:
...
+          println("::notice title=[julia-buildpkg] Caching of the `~/.julia` depot was not detected ::Consider using `julia-actions/cache` to speed up runs https://github.com/julia-actions/cache. To ignore, set input `ignore-no-cache: yes` ")
```
Comment (IanButterworth) [inline action.yml:36]:
```suggestion
          println("::notice title=[julia-buildpkg] Caching of the `~/.julia` depot was not detected ::Consider using `julia-actions/cache` to speed up runs https://github.com/julia-actions/cache. To ignore, set input `ignore-no-cache: true` ")
```

File: action.yml [hunk 4]
```diff
@@ -21,6 +21,9 @@ inputs:
     description: 'Determine if Pkg uses the cli git executable (Julia >= 1.7). Might be necessary for more complicated SSH setups.
                   Options: true | false. Default : false'
     default: 'false'
+  ignore-no-cache:
+    description: 'Whether to ignore if there appears to be no depot caching. Silences an action warning.'
```
Comment (IanButterworth) [inline action.yml:25]:
```suggestion
    description: 'Whether to ignore if there appears to be no depot caching. Silences an action notice recommending `julia-actions/cache`.'
```

**Comment by ViralBShah** [comment]
I don't think I know enough to review this PR. @DilumAluthge Can we disable the review requirements to make maintenance here easier?

**Comment by DilumAluthge** [comment]
I think we can configure Dependbot to auto-approve its own PRs.

File: action.yml [hunk 5]
```diff
@@ -29,8 +32,11 @@ runs:
     run: echo "JULIA_PKG_SERVER_REGISTRY_PREFERENCE=${JULIA_PKG_SERVER_REGISTRY_PREFERENCE:-eager}" >> ${GITHUB_ENV}
     shell: bash
   - run: |
+      if "${{ inputs.ignore-no-cache }}" == "false" && !isdir(DEPOT_PATH[1])
+          println("::notice title=[julia-buildpkg] Caching of the `~/.julia` depot was not detected ::Consider using `julia-actions/cache` to speed up runs https://github.com/julia-actions/cache. To ignore, set input `ignore-no-cache: true` ")
```
Comment (IanButterworth) [inline action.yml:36]:
```suggestion
          println("::notice title=[julia-buildpkg] Caching of the julia depot was not detected ::Consider using `julia-actions/cache` to speed up runs https://github.com/julia-actions/cache. To ignore, set input `ignore-no-cache: true` ")
```

**Review by omus — APPROVED** [review]
LGTM

**Comment by IanButterworth** [comment]
I guess this should pay attention to if JULIA_DEPOT_PATH is set

**Comment by ViralBShah** [comment]
Can we set up dependabot to approve its own PRs?

**Pull request merged**